app/controllers/CategoryGuidanceController.scala

```diff
@@ -38,21 +41,41 @@ class CategoryGuidanceController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   view: CategoryGuidanceView,
   auditService: AuditService,
-  categorisationService: CategorisationService
+  categorisationService: CategorisationService,
+  goodsRecordConnector: GoodsRecordConnector
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
-    with I18nSupport {
+    with I18nSupport
+    with Logging {
 
   def onPageLoad(recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-      request.userAnswers.get(CommodityQuery) match {
-        case Some(_) =>
-          for {
-            _ <- categorisationService.requireCategorisation(request, recordId)
-          } yield Ok(view(recordId))
-        case None    =>
-          Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad().url))
-      }
+      categorisationService
+        .requireCategorisation(request, recordId)
+        .flatMap { userAnswers =>
+          val recordCategorisations = userAnswers.get(RecordCategorisationsQuery)
+          val categorisationInfo    = recordCategorisations.flatMap(_.records.get(recordId))
+          val scenario              = categorisationInfo.map(Scenario.getRedirectScenarios)
+          scenario match {
+            case Some(Category1NoExemptions | StandardNoAssessments) =>
+              CategoryRecord
+                .build(userAnswers, request.eori, recordId)
+                .map { categoryRecord =>
+                  goodsRecordConnector
+                    .updateGoodsRecord(request.eori, recordId, categoryRecord)
+                    .map { _ =>
+                      Redirect(routes.CategorisationResultController.onPageLoad(recordId, scenario.get).url)
```

Comment:
Ideally should go in the navigator but will get swept up in a tech debt ticket anyway so 🤷

Reply:
Thanks for letting me get away with it here, I'm onto something extra now anyway! haha